projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, ValidationErrors, ValidatorFn, Validators } from '@angular/forms';
+import { DataLayerManagerService } from 'ngx-charts-on-fhir';
+
+const BloodPressureRangeValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+  const systolic = control.get('systolic');
+  const diastolic = control.get('diastolic');
+
+  return systolic && diastolic && systolic.value > diastolic.value ? null : { bloodPressure: true };
+};
+
+@Component({
+  selector: 'report-bp',
+  templateUrl: './report-bp.component.html',
+  styleUrls: ['./report-bp.component.css'],
+})
+export class ReportbpComponent implements OnInit {
+  submitted = false;
+  min = 11;
+  max = 250;
+
+  form = this.fb.group(
+    {
+      systolic: [null, [Validators.required, Validators.min(this.min), Validators.max(this.max)]],
+      diastolic: [null, [Validators.required, Validators.min(this.min), Validators.max(this.max)]],
+    },
+    { validator: [BloodPressureRangeValidator] }
+  );
```

Comment:
I'm seeing a warning from Typescript that this overload is deprecated. I think it's because you're missing an `s` in `validators`. This is preventing it from inferring the correct type for the FormGroup, which is probably why you had to use `form.get('diastolic')` instead of `form.controls.diastolic`.
```suggestion
  form = this.fb.group(
    {
      systolic: [null, [Validators.required, Validators.min(this.min), Validators.max(this.max)]],
      diastolic: [null, [Validators.required, Validators.min(this.min), Validators.max(this.max)]],
    },
    { validators: [BloodPressureRangeValidator] }
  );
```